source/vibe/http/common.d

```diff
@@ -449,18 +451,51 @@ final class ChunkedOutputStream : OutputStream {
 	/// ditto
 	@property void maxBufferSize(size_t bytes) { m_maxBufferSize = bytes; if (m_buffer.data.length >= m_maxBufferSize) flush(); }
 
+	/** A delegate used to specify the extensions for each chunk written to the underlying stream.
+	 	
+	 	The delegate has to be of type `string delegate(in const(ubyte)[] data)` and gets handed the
+	 	data of each chunk before it is written to the underlying stream. If it's return value is non-empty,
+	 	it will be added to the chunk's header line.
+
+	 	The returned chunk extension string should be of the format `key1=value1;key2=value2;[...];keyN=valueN`
+	 	and **not contain any carriage return or newline characters**.
+
+	 	Also note that the delegate should accept the passed data through a scoped argument. Thus, **no references
+	 	to the provided data should be stored in the delegate**. If the data has to be stored for later use,
+	 	it needs to be copied first.
+	 */
+	@property ChunkExtensionCallback chunkExtensionCallback() const { return m_chunkExtensionCallback; }
+	/// ditto
+	@property void chunkExtensionCallback(ChunkExtensionCallback cb) { m_chunkExtensionCallback = cb; }
+
+	private void append(scope void delegate(scope ubyte[] dst) del, ulong nbytes)
```

Comment:
Okay, thanks again for the quick responses and extension of `AllocAppender`!
I integrated the `AllocAppender.append()` calls and squashed everything into a single commit. To homogenize the buffer handling in both `write()` overloads I implemented a new (private) `append()` method. 

Do you think there's further need for modifications (any obvious gotchas)?